### ISSUE TITLE
Can't add instrument  to B+B editor when B+B has no vertical space left

### ISSUE THREAD

**Issue by karmux** [LMMS/lmms#3099]
I have many samples and intruments in B+B editor and there is no vertical space left on the screen. I didn't find a way to add instrument plugins from left side to B+B editor. Only way is to drag and release on existing instrument and replacing it.

Samples and presets can be added by right click and Open in new instrument-track/B+B Editor. I believe this context menu can be useful for instrument plugins also.

**Comment by BaraMGB** [comment]
I've never noticed that there isn't that functionality in instruments. I'll look into this.

**Comment by DeRobyJ** [comment]
Basically we don't have right-click options on plugin instruments (outside of the browser), and we don't have a space in BBE or SE to drop things, if they are "full".

A good fix for this would be to add the ability to drop things in the whole window of SE and BBE (for instanc,e on the title or the buttons bar), not just on empty spaces.
Or we can just have right click on that browser too, what's more intuitive?

(a lazy fix, which is even more intuitive for the user, would be to have extra empty space, like half a track, on top or bottom of the tracks)

**Comment by musikBear** [comment]
@DeRobyJ  wrote
> and we don't have a space in BBE or SE to drop things, if they are "full".

Its not entirely true, but it is weirdly implemented. You need to 'drop' on the *slider*
That is ofcause awkward and not intuitive at all, but it works with both instruments and presets.
But i would support consistency in this respect, and add a context menu on the **instruments** just as we have it for the presets, and samples.

**Comment by tresf** [comment]
> You need to 'drop' on the slider

Drag to play button works too.  @musikBear and I drew up this mock-up when I believe would help.... https://github.com/LMMS/lmms/issues/1531

**Comment by musikBear** [comment]
@tresf Yes nice, but since the dragging 'kind off' induce an additional event, namely the pre-hear, then imho all dragging should better be replaced with informative *context-menus*. Dragging is a leftover from earliest versions, and it is not precise, and error-prone. (Here is one that had crash when he dragged a zasfx: https://lmms.io/forum/viewtopic.php?f=7&t=5867&p=21886#p21886. -But that could also be other reasons.

**Comment by Spekular** [comment]
Personally I don't really see anything wrong with dragging, so I think we should keep it. I find it very uninterruptive of my "flow" to drag as oppised to using a context menu. I never want to add a sample or instrument to the BB editor without hearing it first, so the preview isn't an issue either.

That said the option of a context menu is great, I just wouldn't consider it a replacement for dragging.

**Comment by Umcaruje** [comment]
This one has been bugging me as well :+1:
